enhancements/multi-arch/heterogenous-architecture.md

```diff
@@ -0,0 +1,270 @@
+capability of a cluster on different topologies. The phases are listed below and details on some components of the phases will be described below. 
+
+Phase 0: Prototype
+- Aimed at delivering a prototype that users can experiment with
+- Aimed at Azure/AWS with the primary supported architectures being x86 + ARM.
+- For supporting ARM nodes on Azure, [hyper-v support][BZ-1949614] is required which is landing in RHEL 8.6 and will be included in RHCOS 4.11
+- oc adm command changes to support manifest list payloads
+- Ability to assemble and publish manifest listed release payloads in a reproducible way
+- Does not include upgrade support
+- Is not intended for production
+
+Phase 1: GA of basic support
+- Release pipeline needs to be setup to construct manifest list based release payloads
+- Upgrade support - CVO/Cincinnati support (heterogenous -> heterogenous payload upgrades only)
+- Does not include support for:
+    - image registry - internal image registry will not support manifest listed images
+    - imagestreams
+
+Phase 2: Full basic functionality
+- Imagestream manifest list support
+- OLM/Operator ecosystem improvements
+- oc debug (and other commands) support for compute architecture
+- support toolbox
+- support for Power and Z architectures
+- Hypershift awareness
+
+Phase 3: Quality of life improvements
+- First class install support - create workers at install time
+- Improve console's content filtering by making it architecture aware
+- Possibly reduce mirroring costs by looking into things like sparse manifest list
+- Upgrade support for homogenous payload -> heterogenous payload and vice versa
+
+### User Stories
+
+As Openshift continues to add to the architecture types that it is able to run on (x86, Power, Z, ARM) it is becoming more obvious that there are clear requirements and use cases for Openshift clusters that contain mixed architecture types.
+This becomes even more of the case once we introduce ARM support as a significant number of use cases detail a x86 master node talking to ARM worker nodes. While ACM could be an option in some cases it does not cover the main ones
+we are seeing that this area would need to address.
+
+### API Extensions
+
+Upgrades to a heterogenous payload will require a special field either in the infrastructure or the cluster version resource to indicate that the user wants to upload to a heterogenous image. One of these APIs will be affected to include this.
+
+### Implementation Details/Notes/Constraints
+
+Operators supporting multiple architectures cannot stop supporting a given architecture since it would break heterogenous clusters which are trying to upgrade
+
+### Risks and Mitigations
+
+Releases and upgrade graphs for all architectures have to move simultaneously and cannot be independent. If there is a bug on one arch, it will cause the edge or the release image to be removed for all arches
```

Comment:
is this really true?  We can't ship a homogeneous payload if any of the heterogeneous payloads are blocked, but we can still ship the independent heterogeneous payloads(for the unblocked architectures) if we want to, right?

Reply:
ok. you mean allow shipping of the heterogeneous payload for only architectures that are not blocked? how would we enforce that? through a support statement? or not including the payload for the blocked arch?

Reply:
> you mean allow shipping of the heterogeneous payload for only architectures that are not blocked? 

no, i mean that if we are about to release 4.11.3, and we determine that 4.11.3 is broken for ARM, we can still ship the 4.11.3 homogeneous payloads for x86, power, and s390.  We cannot ship the ARM payload, and we cannot ship the heterogeneous payload (since it would be missing ARM, or have a different version of ARM).

heterogeneous depends on homogeneous, but each individual homogeneous should be independent/able to ship.

Reply:
ok yeah that makes sense. so from an upgrade perspective, the edges for the particular architecture's homogeneous payload would be missing and should have no other implications i guess.

Reply:
actually today we do hold up a release if one architecture is broken because we would have to block that arch in Cincinnati ..so i guess in that sense if one arch breaks, no releases ship.

Reply:
updated this statement and made it simpler.